game/fnaacm/bots/dumb_bot.py

```diff
@@ -14,10 +13,24 @@ def __init__(self):
         self.stun = False
 
     @override
-    def _calc_next_move_patrol(self, gameboard : GameBoard, player: Player) -> list[ActionType]:
+    def __calc_next_move_patrol(self, gameboard : GameBoard, player: Player) -> list[ActionType]:
         return self.movement()
 
     @override
+    def __calc_next_move_hunt(self, gameboard : GameBoard, player : Player) -> list[ActionType]:
+        route = self.player_seen_movement(player)
+        execute_action = [route[0]]
+        return execute_action
+
+    @override
+    def calc_next_move(self, gameboard : GameBoard, player : Player) -> list[ActionType]:
+        """
+        returns actions that the bot should take to get to wherever it wants to go (typically player vector)
+        """
+        if self.can_see_player(gameboard, player):
+            return self.__calc_next_move_hunt(gameboard, player)
+        return self.__calc_next_move_patrol(gameboard, player)
+    @override
     def _calc_next_move_hunt(self, gameboard : GameBoard, player : Player) -> list[ActionType]:
         return self.player_seen_movement(player)
```

Comment:
to return a new list only containing the first element (as in your implementation above), use list slicing
```suggestion
        return self.player_seen_movement(player)[:1]
```
https://www.geeksforgeeks.org/python/python-list-slicing/#